### PR TITLE
Feature/builtin errors

### PR DESCRIPTION
**Pull request by shiblon**
tl;dr: race condition fixed in handling of MoveTaskError and friends, moved from pkg/errors to built-in errors/fmt, and moved all backend code into the backend directory. This is a breaking change.

This is a huge change. It removes all instances of `github.com/pkg/errors` and replaces all uses of that package with the built-in error handling and formatting, namely

- `errors.Wrap`* -> `fmt.Errorf` with the `%w` format specifier
- `errors.Cause` -> `errors.Is` and `errors.As`.

The second change was tricky in a couple of places where the original cause was needed and we didn't have an actual type to check for. This happens in cases where we are looking for an interface type, which might just need to change at some point to not be so clunky.

All tests pass.

In addition to the above, a race condition was found and fixed that occasionally triggered a failed `WorkerMoveOnError` test. The race was fixed in a late commit on this branch, and details can be found there. The symptom of this is that, occasionally when a task has been renewed before being moved with a `MoveTaskError`, it would cause the worker to die instead of moving the task. That has been fixed.

Finally, all backend code is now in the `backend` directory, and backend packages have all been renamed to have the `eq` prefix in the name to avoid collisions with standard packages (e.g., `grpc`).

### CODE REVIEW

**Review by ButchIstook — APPROVED** [review]
I read through all the changes, and in general they look great. I am still working to understand the subtlety in the race-condition fixes :)
